diff --git a/lvdm/modules/encoders/condition.py b/lvdm/modules/encoders/condition.py
--- a/lvdm/modules/encoders/condition.py
+++ b/lvdm/modules/encoders/condition.py
@@ -262,9 +262,12 @@ class FrozenOpenCLIPEmbedder(AbstractEncoder):
     def encode_with_transformer(self, text):
         x = self.model.token_embedding(text)  # [batch_size, n_ctx, d_model]
         x = x + self.model.positional_embedding
-        x = x.permute(1, 0, 2)  # NLD -> LND
+        batch_first = getattr(self.model.transformer, "batch_first", False)
+        if not batch_first:
+            x = x.permute(1, 0, 2)  # NLD -> LND
         x = self.text_transformer_forward(x, attn_mask=self.model.attn_mask)
-        x = x.permute(1, 0, 2)  # LND -> NLD
+        if not batch_first:
+            x = x.permute(1, 0, 2)  # LND -> NLD
         x = self.model.ln_final(x)
         return x
 
